src/backend/distributed/sql/citus--11.0-4--11.1-1.sql

```diff
@@ -111,7 +111,8 @@ CREATE TABLE citus.pg_dist_background_tasks(
     not_before timestamptz, -- can be null to indicate no delay for start of the task, will be set on failure to delay retries
     message text,
 
-    CONSTRAINT pg_dist_background_tasks_pkey PRIMARY KEY (job_id, task_id)
+    CONSTRAINT pg_dist_background_tasks_pkey PRIMARY KEY (job_id, task_id),
+    UNIQUE(task_id)
```

Comment:
Maybe this should be the PK instead.

Reply:
I chose the PK to be `(job_id, task_id)`, one could argue both ways I think. Tasks, and their depending tasks, should all be within one job. Meaning we would want to identify the task by jobid, taskid. This prevents any tasks starting to be referenced outside of its job, both in sql and in C.

task id's only recently had a hard requirement to be unique, because we want to lock on them.

I am fine with swapping them around, although then it's not a unique constraint anymore, but instead, just an index on jobid/taskid.

Bit of a problem would be to redo all the C code that assumes the task is identified by `jobid, taskid` but still doable if concensus is to change the primary key here.